docs/markdown/Reference-manual.md

```diff
@@ -999,6 +1002,9 @@ This is useful for building modules that will be `dlopen()`ed and
 hence may contain undefined symbols that will be provided by the
 library that is loading it.
 
+If you want an [`executable`](#executable) to be able to load shared
+modules, you will need to set its `modular` argument to `true`.
+
```

Comment:
This is not a good description of what this argument does, and `man ld` manages better.

`-export-dynamic` is _only_ needed if you want to load a shared module _which references symbols in the executable_.

Given that, the name `modular` for this argument is not well-chosen.

See test case common/156 (which uses the compiler option `-rdynamic `which causes this linker option to be used), which should probably be updated to use this argument. (and compare with test case common/125, which does not use or need this option)